xbmc/platform/posix/filesystem/SMBDirectory.cpp

```diff
@@ -121,137 +107,107 @@ bool CSMBDirectory::GetDirectory(const CURL& url, CFileItemList &items)
+    if (S_ISREG(st.st_mode) || bIsDir)
+    {
+      nstats++;
+      // This is also present in stuct stat using S_IXOTH but given the potential for confusion
+      // let's be explicit.
+      hidden = hidden || fi->attrs & SMBC_DOS_MODE_HIDDEN;
+
+      lTimeDate = st.st_mtime;
+      if (lTimeDate == 0) // if modification date is missing, use create date
+        lTimeDate = st.st_ctime;
+      iSize = st.st_size;
+    }
 
-        struct stat info = {};
-        if ((m_flags & DIR_FLAG_NO_FILE_INFO)==0 && CServiceBroker::GetSettingsComponent()->GetAdvancedSettings()->m_sambastatfiles)
-        {
-          // make sure we use the authenticated path which contains any default username
-          const std::string strFullName = strAuth + smb.URLEncode(strFile);
-
-          lock.lock();
-          if (!smb.IsSmbValid())
-          {
-            items.ClearItems();
-            return false;
-          }
-
-          nstats++;
-          if( smbc_stat(strFullName.c_str(), &info) == 0 )
-          {
-
-            char value[20];
-            // We poll for extended attributes which symbolizes bits but split up into a string. Where 0x02 is hidden and 0x12 is hidden directory.
-            // According to the libsmbclient.h it returns 0 on success and -1 on error.
-            // But before Samba 4.17.5 it seems to return the length of the returned value
-            // (which is 4), see https://bugzilla.samba.org/show_bug.cgi?id=14808.
-            // Checking for >= 0 should work both for the old erroneous and the correct behaviour.
-            ngetattr++;
-            if (smbc_getxattr(strFullName.c_str(), "system.dos_attr.mode", value, sizeof(value)) >= 0)
-            {
-              long longvalue = strtol(value, NULL, 16);
-              if (longvalue & SMBC_DOS_MODE_HIDDEN)
-                hidden = true;
-            }
-            else
-              CLog::Log(
-                  LOGERROR,
-                  "Getting extended attributes for the share: '{}'\nunix_err:'{:x}' error: '{}'",
-                  CURL::GetRedacted(strFullName), errno, strerror(errno));
-
-            bIsDir = S_ISDIR(info.st_mode);
-            lTimeDate = info.st_mtime;
-            if(lTimeDate == 0) // if modification date is missing, use create date
-              lTimeDate = info.st_ctime;
-            iSize = info.st_size;
-          }
-          else
-            CLog::Log(LOGERROR, "{} - Failed to stat file {}", __FUNCTION__,
-                      CURL::GetRedacted(strFullName));
-
-          lock.unlock();
-        }
-      }
+    KODI::TIME::FileTime fileTime, localTime;
+    KODI::TIME::TimeTToFileTime(lTimeDate, &fileTime);
+    KODI::TIME::FileTimeToLocalFileTime(&fileTime, &localTime);
 
-      KODI::TIME::FileTime fileTime, localTime;
-      KODI::TIME::TimeTToFileTime(lTimeDate, &fileTime);
-      KODI::TIME::FileTimeToLocalFileTime(&fileTime, &localTime);
+    if (bIsDir)
+    {
+      CFileItemPtr pItem(new CFileItem(strFile));
```

Comment:
Instead of the legacy typedef use the real type `std::make_shared<CFileItem>`

Reply:
What I meant here is `auto item = std::make_shared<CFileItem>(strFile);` Also make sure you drop the `p` prefixes